packages/block-library/src/button/edit.js

```diff
@@ -61,86 +93,104 @@ class ButtonEdit extends Component {
 			fallbackBackgroundColor,
 			fallbackTextColor,
 			setAttributes,
-			isSelected,
 			className,
+			instanceId,
+			isSelected,
 		} = this.props;
 
 		const {
 			text,
 			url,
 			title,
+			linkTarget,
+			rel,
 		} = attributes;
 
+		const linkId = `wp-block-button__inline-link-${ instanceId }`;
+
 		return (
-			<>
-				<div className={ className } title={ title } ref={ this.bindRef }>
-					<RichText
-						placeholder={ __( 'Add text…' ) }
-						value={ text }
-						onChange={ ( value ) => setAttributes( { text: value } ) }
-						formattingControls={ [ 'bold', 'italic', 'strikethrough' ] }
-						className={ classnames(
-							'wp-block-button__link', {
-								'has-background': backgroundColor.color,
-								[ backgroundColor.class ]: backgroundColor.class,
-								'has-text-color': textColor.color,
-								[ textColor.class ]: textColor.class,
-							}
-						) }
-						style={ {
-							backgroundColor: backgroundColor.color,
-							color: textColor.color,
-						} }
-						keepPlaceholderOnFocus
+			<div className={ className } title={ title } ref={ this.bindRef }>
+				<RichText
+					placeholder={ __( 'Add text…' ) }
+					value={ text }
+					onChange={ ( value ) => setAttributes( { text: value } ) }
+					formattingControls={ [ 'bold', 'italic', 'strikethrough' ] }
+					className={ classnames(
+						'wp-block-button__link', {
+							'has-background': backgroundColor.color,
+							[ backgroundColor.class ]: backgroundColor.class,
+							'has-text-color': textColor.color,
+							[ textColor.class ]: textColor.class,
+						}
+					) }
+					style={ {
+						backgroundColor: backgroundColor.color,
+						color: textColor.color,
+					} }
+					keepPlaceholderOnFocus
+				/>
+				<BaseControl
+					label={ __( 'Link' ) }
+					className="wp-block-button__inline-link"
+					id={ linkId }>
+					<URLInput
+						className="wp-block-button__inline-link-input"
+						value={ url }
+						onChange={ ( value ) => setAttributes( { url: value } ) }
+						showSuggestionsOverride={ ! isSelected ? false : undefined }
```

Comment:
I'm thinking an `enableSuggestions` or `disableSuggestions` prop might be easier to grok and to document?

```js
enableSuggestions={ isSelected }
```

Reply:
There's a reason I didn't go for something like that. `showSuggestionsOverride` can have three states—true, to always show suggestions, false to never show suggestions, or undefined to fallback to the implicit behaviour for showing suggestions.

A prop name of either `enableSuggestions` or `disableSuggestions` only indicates the two boolean states. `enableSuggestions={ undefined }` conveys that suggestions would be disabled.

It is frustratingly more complex than I would've hoped.

Reply:
Decided to go with `disableSuggestions` in #16497, which I think covers things.